pkg/compiler/symbol_table_test.go

```diff
@@ -0,0 +1,45 @@
+package compiler
+
+import "testing"
+
+func TestDefine(t *testing.T) {
+	expected := map[string]Symbol{
+		"a": {Name: "a", Scope: GlobalScope, Index: 0},
+		"b": {Name: "b", Scope: GlobalScope, Index: 1},
```

Comment:
Would we want a symbol a with an already used name, e.g. `"a"`?

Reply:
I meant: should we add a test case that has two symbols with the same name, just to address the concerns raised earlier: should entries in a symbol table ever be overridden. Maybe all a bit out of context, feel free to ignore.